### PR TITLE
Dockerfile + github workflows

### LINKED ISSUE #16: Create the dockerfile to deploy the fertiscan backend
TODO:

- Create a dockerfile

### LINKED ISSUE #19: Add github workflows
Should include:

- [x] Docker build and push to ghcr
- [x] Python linting
- [ ] #24
- [x] ai-cfia repo standard
- [x] Markdown check

### PR DESCRIPTION
**Pull request by ThomasCardin**
- Closes #16 
- Closes #19

### CODE REVIEW

**Comment by ThomasCardin** [comment]
@snakedye @SamuelPelletierEvraire Hey guys, could you fix all errors prompt by the github workflows :D Thanks

**Comment by ThomasCardin** [comment]
Workflow `python-coverage` won't be added. It's still under construction

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+COPY . .
+
+RUN pip install --no-cache-dir -r requirements.txt
+
+ARG ARG_AZURE_API_ENDPOINT
+ARG ARG_AZURE_API_KEY
+ARG ARG_AZURE_OPENAI_API_ENDPOINT
+ARG ARG_AZURE_OPENAI_API_KEY
+ARG ARG_PROMPT_PATH
+ARG ARG_UPLOAD_PATH
```
Comment (snakedye) [inline Dockerfile:14]: Add this [envar ](https://github.com/ai-cfia/fertiscan-backend/blob/4c5a11fe51e0703566989be547c04e240057962d/.env.template#L12) to the list.

File: README.md [hunk 2]
```diff
@@ -5,19 +5,51 @@ A flask-based backend for FertiScan.
...
+
+    ```bash
+    docker build -t fertiscan-backend \
+    --build-arg ARG_AZURE_API_ENDPOINT=your_actual_azure_form_recognizer_endpoint \
+    --build-arg ARG_AZURE_API_KEY=your_actual_azure_form_recognizer_key \
+    --build-arg ARG_AZURE_OPENAI_API_ENDPOINT=your_actual_azure_openai_endpoint \
+    --build-arg ARG_AZURE_OPENAI_API_KEY=your_actual_azure_openai_key \
+    --build-arg ARG_PROMPT_PATH=actual_path/to/prompt_file \
+    --build-arg ARG_UPLOAD_PATH=actual_path/to/upload_file \
+    .
```
Comment (snakedye) [inline README.md:36]: Same as the previous one. Add this [variable](https://github.com/ai-cfia/fertiscan-backend/blob/4c5a11fe51e0703566989be547c04e240057962d/.env.template#L12).

**Review by snakedye — CHANGES_REQUESTED** [review]
GG. Just a few details to address and it's good.

File: Dockerfile (hunk 1 above)
Comment (ThomasCardin) [reply Dockerfile:14]: Good I will also update the README.md

File: README.md (hunk 2 above)
Comment (ThomasCardin) [reply README.md:36]: Good I will also update the README.md

**Pull request merged**